### PR TITLE
frontend: make unannotated asm blocks error instead of deprecation

### PR DESCRIPTION
**Pull request by ljmf00**
This is a really old deprecation that already passed the deprecation period for
more than 5 years.

Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

**Comment by dkorpel** [comment]
Tests need to updated, and a changelog entry should be added.

**Comment by ljmf00** [comment]
> Tests need to be updated, and a changelog entry should be added.

I think the spec doesn't mention this deprecation. Should I update it?

File: src/dmd/statementsem.d [hunk 1]
```diff
@@ -3922,9 +3922,9 @@ package (dmd) extern (C++) final class StatementSemanticVisitor : Visitor
         // use setImpure/setGC when the deprecation cycle is over
```
Comment (dkorpel) [inline src/dmd/statementsem.d:3922]: As the comments says, instead of raising an error, you can just set the function to be impure / gc

Comment (ljmf00) [reply src/dmd/statementsem.d:3922]: Ah right. Thanks, going to fix it :)

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Is this good to go?

**Comment by ljmf00** [comment]
> Is this good to go?

Yes, just need to write a note on deprecated.dd on spec.

File: changelog/deprecation_asm_stc.dd [hunk 2]
```diff
@@ -0,0 +1,6 @@
+The deprecation period of unannotated `asm` blocks has been ended.
+
+See the $(LINK2 $(ROOT_DIR)deprecate.html#asm, Deprecated Features) for more information.
+
+Starting with this release, using unannotated `asm` blocks will result in a
```
Comment (dkorpel) [inline changelog/deprecation_asm_stc.dd:5]: They won't result in a compiler error per se, they'll be assumed to be the opposite of `pure` `@safe` `nothrow` or `@nogc.

**Review by dkorpel — CHANGES_REQUESTED** [review]

**Review by dkorpel — APPROVED** [review]

File: changelog/deprecation_asm_stc.dd (hunk 2 above)
Comment (ljmf00) [reply changelog/deprecation_asm_stc.dd]: Changed the wording

**Pull request merged**